framer/EventEmitter.coffee

```diff
@@ -1,3 +1,94 @@
-EventEmitter = require "eventemitter3"
```

Comment:
Yeah, but event emitters are pretty easy, the code is well tested, and I've written them many times before.

Maybe this version a tiny bit slower because I don't do fancy case statements per n arguments, but we can optimize if it turns out to be a problem, and it really doesn't seem to be the case. Other than that, they are mostly the same.

I really think the tradeoff is worth it, having a better removal api, being able to remove the layer cruft, and having some sanity checking built in are good things.